chain/client/src/test_utils.rs

```diff
@@ -1102,7 +1102,151 @@ pub struct TestEnv {
     pub clients: Vec<Client>,
 }
 
+/// A builder for the TestEnv structure.
+pub struct TestEnvBuilder {
+    chain_genesis: ChainGenesis,
+    clients: Vec<AccountId>,
+    validators: Vec<AccountId>,
+    runtime_adapters: Option<Vec<Arc<dyn RuntimeAdapter>>>,
+    network_adapters: Option<Vec<Arc<MockNetworkAdapter>>>,
+}
+
+/// Builder for the `TestEnv` structure.
```

Comment:
nitpick: we might start using `` [`TestEnv`] `` syntax to get proper links in rustdoc and goto definition in IDE.

Reply:
Yeah, good point. I was actually write that but then I forgot whether that’s actually rustdoc syntax and then forgot to check it and correct it… I’ll fix that (and potentially other issues) in separate commit.